peers/app_request_network.go

```diff
@@ -103,44 +103,33 @@ func NewNetwork(
 		)
 		return nil, nil, err
 	}
-	ip, err := infoClient.GetNodeIP(context.Background())
-	if err != nil {
-		logger.Error(
-			"Failed to get ip",
-			zap.Error(err),
-		)
-		return nil, nil, err
-	}
-	id, _, err := infoClient.GetNodeID(context.Background())
-	if err != nil {
-		logger.Error(
-			"Failed to get node id",
-			zap.Error(err),
-		)
-		return nil, nil, err
-	}
-	beaconIPs = append(beaconIPs, ip)
-	beaconIDs = append(beaconIDs, id.String())
-
-	var indices []uint64
-	// If we have more peers than numInitialTestPeers, take a random sample of numInitialTestPeers peers
-	if len(peers) > numInitialTestPeers {
-		s := sampler.NewUniform()
-		s.Initialize(uint64(len(peers)))
-		indices, _ = s.Sample(numInitialTestPeers)
-	} else {
-		for i := range peers {
-			indices = append(indices, uint64(i))
-		}
-	}
 
+	// Randomly select peers to connect to until we have numInitialTestPeers
+	indices := rand.Perm(len(peers))
 	for _, index := range indices {
 		// Do not attempt to connect to private peers
 		if len(peers[index].PublicIP) == 0 {
 			continue
 		}
 		beaconIPs = append(beaconIPs, peers[index].PublicIP)
 		beaconIDs = append(beaconIDs, peers[index].ID.String())
+		if len(beaconIDs) == numInitialTestPeers {
+			break
+		}
+	}
+	if len(beaconIPs) == 0 {
+		logger.Error(
+			"Failed to find any peers to connect to",
+			zap.Error(err),
+		)
+		return nil, nil, err
+	}
+	if len(beaconIPs) < numInitialTestPeers {
```

Comment:
why do we need to limit the number of peers to 5 again? IIUC when we do the signature aggregation process it could take longer in terms of us sending out more signature requests, but would increase likelihood of having enough stake threshold in the peers. Are 5 peers generally enough for exceeding stake threshold?

Reply:
This is just the list of initial peers to manually connect to in order to initialize the app request network. When requesting signatures, we connect to all peers from which we are requesting signatures.